ui/src/features/project/pipelines/create-warehouse-modal.tsx

```diff
@@ -24,7 +26,16 @@ const formSchema = z.object({
 
 export const CreateWarehouseModal = ({ visible, hide, project }: Props) => {
   const { mutateAsync, isPending } = useMutation(createResource, {
-    onSuccess: () => hide()
+    onSuccess: (response) => {
+      for (const result of response?.results || []) {
+        if (result?.result?.case === 'createdResourceManifest') {
+          queryCache.warehouse.add(project, [
+            decodeUint8ArrayYamlManifestToJson(result?.result?.value)
+          ]);
+        }
+      }
```

Comment:
Why is this needed if we have a watcher for Warehouses?

Reply:
I think you are right. I resolved a bug with watch after I added this invalidation, now we don't need this anymore.